src/server/modules/api/external-v1/index.ts

```diff
@@ -0,0 +1,24 @@
+import { StorableUrl } from '../../../repositories/types'
+
+export { ApiController } from './ApiController'
+
+type OptionalLongUrlProperty = {
+  longUrl?: string
+}
+type UserIdProperty = {
+  userId: number
+}
+
+type ShortUrlProperty = {
+  shortUrl: string
+}
+
+type ShortUrlOperationProperty = UserIdProperty & ShortUrlProperty
+
+export type UrlCreationRequest = ShortUrlOperationProperty &
+  OptionalLongUrlProperty
+
+export type ApiUrl = Pick<
```

Comment:
May be UrlV1DTO, 🤔

Reply:
done in [07bce66](https://github.com/opengovsg/GoGovSG/pull/2069/commits/07bce665fac53138bbc48a6b793465c409637ecf)